wrappers/python/openmm/app/simulation.py

```diff
@@ -207,8 +207,14 @@ def _simulate(self, endStep=None, endTime=None):
             anyReport = False
             for i, reporter in enumerate(self.reporters):
                 nextReport[i] = reporter.describeNextReport(self)
-                if nextReport[i][0] > 0 and nextReport[i][0] <= nextSteps:
-                    nextSteps = nextReport[i][0]
+                # Allow to pass either a dictionary, or a tuple from the reporter for backwards compatibility
```

Comment:
If the return value is a tuple, let's immediately convert it to the new format.  Then we don't have to worry about alternate formats in anything that follows.  Something like this (not tested, I'm just writing this in the browser).

```python
report = reporter.describeNextReport(self)
if isinstance(report, tuple):
    # Convert an old format tuple to the new dict format
    includes = ['positions', 'velocities', 'forces', 'energy']
    include = [includes[i] for i in range(4) if report[i+1]]
    newReport = {'steps': report[0], 'include':include}
    if len(report) > 5:
        newReport['periodic'] = report[5]
    report = newReport
nextReport[i] = report
```